rust/property-defs-rs/src/config.rs

```diff
@@ -36,6 +36,17 @@ pub struct Config {
     #[envconfig(default = "1000000")]
     pub cache_capacity: usize,
 
+    // We impose a slow-start, where each batch update operation is delayed by
+    // this many ms times the % of remaining cache space (so with a 50% full cache,
+    // we delay by e.g. 500ms, at 90% full, we delay by 100ms)
```

Comment:
Can you say more about why? Reading this comment I'm not sure what purpose this is actually serving, and my guess is future readers will feel the same, when I'm sure there's a good reason for the delay.

Reply:
I'll revisit the comment, you're right that it doesn't explain things well. My thinking was this:

`writes/s = (batch size * concurrency_limit) / (batch write time)` - e.g. with concurrency of 4, batch size of 10, batches write time of half a second: `(4*10)/0.5 = 80 writes/second`. Arbitrarily delaying at write time imposes an upper limit of `(batch size * concurrent writes / delay)`, even if writes were instant, which at our current settings is 50_000/s (concurrency of 10, batch size of 5000, delay time of 1 second when cache is empty). You want to slowly raise that artificial limit until you get rid of it entirely, as the filtering starts to work (if you've tuned your cache size so that enough filtering is done that you don't exceed a safe write rate)

You could make this an absolute limit by calculating "I want to write at most X/second", then backsolving for what write time should be, and delaying by `calculated insert time - actual insert time` after doing the insert, but that's being a little fancy I think (you start having to also adjust for batch size and I just don't really want to do the maths)